### PR TITLE
Bug Fix e2e tests

### LINKED ISSUE #384: [BUG] e2e tests are flaky
**Describe the bug**

e2e tests on CI are failing for update events validate

See run history on master - e.g https://github.com/infracloudio/botkube/runs/1193896198

### PR DESCRIPTION
**Pull request by codenio**
This Commit, 
- Fixes build failure due to e2e tests

##### ISSUE TYPE

 - Bug fix Pull Request

##### SUMMARY

Fixes #384

### CODE REVIEW

**Review by sanketsudake — APPROVED** [review]
I assume these changes are for further debugging of tests failure.

File: test/e2e/utils/utils.go [hunk 1]
```diff
@@ -104,11 +105,15 @@ func UpdateResource(t *testing.T, obj UpdateObjects) (*unstructured.Unstructured
 	if err != nil {
 		t.Fatalf("Failed to create %s: %v", obj.GVR.Resource, err)
 	}
+	// Provide Sufficient time interval for messages to reach fack servers
+	time.Sleep(30 * time.Second)
```
Comment (PrasadG193) [inline test/e2e/utils/utils.go:109]: Adding sleep is not reliable and cleaner way.
We should have a retry logic with timeout to wait till the condition met

**Comment by codenio** [comment]
> I assume these changes are for further debugging of test failure.

yes @sanketsudake, The test cases are working fine in local, so trying to check through Github actions and debug further.
we need to hold this for now.

File: test/e2e/utils/utils.go (hunk 1 above)
Comment (codenio) [reply test/e2e/utils/utils.go:109]: trying out possibilities to understand the problem with GitHub actions, since there is no issues with local testing.

Comment (codenio) [reply test/e2e/utils/utils.go:109]: Now it is assured that it is due to lack of time for messages reaching fack servers, will work on logics and update here

**Comment by codenio** [comment]
Tried different possible ways of fixing this issue. Doing so, we can understand that, we need to Mock the time delay involved in listening, filtering, and notifying events to all notifiers or we need to test each notifier individually

@PrasadG193 @sanketsudake thoughts and comments on this..?

**Comment by PrasadG193** [comment]
@codenio let's merge this fix for now so that other PRs will get unblocked. We will definitely have to revisit this and come up with a better solution

**Review by PrasadG193 — APPROVED** [review]

**Comment by chetanpdeshmukh** [comment]
Hello @codenio ,
Thanks again for the PR!

Thank you for contributing to BotKube. Could you please fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSdULT22oZlpHPMXKVEtV4fZ6R8vdGDC2iFvMhoG6IhZkK1SFA/viewform?usp=sf_link), so we can send the well deserved awesome swags :slightly_smiling_face:

Team BotKube

**Pull request merged**